satpy/readers/li_base_nc.py

```diff
@@ -439,11 +439,12 @@ def get_measured_variable(self, var_paths, fill_value=np.nan):
         # Also handle fill value here (but only if it is not None, so that we can still bypass this
         # step if needed)
         arr = self.apply_fill_value(arr, fill_value)
-
         return arr
 
     def apply_fill_value(self, arr, fill_value):
-        """Apply fill values, unless it is None."""
+        """Apply fill values, unless it is None and when _FillValue is provided."""
+        if arr.attrs.get("_FillValue") is None:
+            return arr
         if fill_value is not None:
```

Comment:
I think the logic could be simplified to this, if you agree
`if fill_value is not None and arr.attrs.get("_FillValue") is not None:`